Feature/serv-1000 hello jwp app

AmitaiB opened the pull request:
### What does this Pull Request do?
1. It refactors the Swift and ObjC demo view controllers for simplicity, while maintaining outward behavior and programmatic demonstration of features.
2. It establishes parity between the Swift and ObjC implementations -- an integrator familiar with one language can see the same abstraction implemented in either language.
3. It normalizes the project/target/repo name(s).
    * Overwrites the project and directory names with the repo name: before, each had a distinct name. Now they are all '**JWPlayer-SDK-iOS-Demo**' (ignoring capitalization).
4. It updates the Swift code to 4.2
5. Incorporates @karimMourra 's offline comments:
    * the focus is narrowed, and the abstraction is tightened from "A [QuickStartGuide](https://en.wikipedia.org/wiki/Quickstart_guide) App" to "Simplest Concise Implementation of an Approachable Minimal Player Instance":
     * code implementing `JWPlayerDelegate` callbacks are removed.
    * code implementing `JWPlayerNotification` s are removed.

### Why is this Pull Request needed?
Addresses internal requests from Account Management, Sales, and Support members, respectively, for a straightforward demo that was current, and put our best foot forward.

### Are there any points in the code the reviewer needs to double check?
Nothing in the version to be reviewed.

However, just to avoid confusion, I will mention that the update originally remade the app as a streamlined, 'HelloJWP_MVVM' demo, so as to separate the SDK code from the UI code. 

This changes after commit 05394d77224d4d9cf743d58d30c911f5b7d54352 ("Simplifies demo to minimal viable player + captions + ads"), when I incorporate the principles of point 5 (above). 

### Are there any Pull Requests open in other repos which need to be merged with this?
No.

#### Addresses Issue(s):
SERV-1000

AmitaiB committed:
Updates to SDK v3.2.3

AmitaiB committed:
Project Settings update

AmitaiB committed:
Impls SwiftViewModel for new VC

AmitaiB committed:
Impls SwiftViewController to replace SwiftPlayerViewController
squash

AmitaiB committed:
sq - begins impl ObjC VC

AmitaiB committed:
Impls AVAudioSession set to playback in AppDelegate

AmitaiB committed:
Pauses player in viewDidDisappear

AmitaiB committed:
Cleans up ViewModel

AmitaiB committed:
Implements ObjC View Controller
squash

AmitaiB committed:
Refactors Endpoints and JWPlayer creation out of bloated VM
squash

AmitaiB committed:
Converts to Swift 4.2

AmitaiB committed:
Aligns project name with directory's name
squash

AmitaiB committed:
Final pre-simplify version

AmitaiB committed:
Merge branch 'feature/SERV-1000-update-demo-app' into develop

AmitaiB committed:
Simplifies demo to minimal viable player + captions + ads

Removes MVVM architecture, console output, JWPlayerDelegate callbacks, and associated properties, classes, and helpers.

AmitaiB committed:
Updates to SDK 3.2.4, aligns Display Name, default video loads faster

AmitaiB committed:
Fixes LaunchScreen images

AmitaiB committed:
Restores AppIcon set

karimMourra commented on JWPlayer-SDK-iOS-Demo/AppDelegate.m:
```diff
@@ -0,0 +1,37 @@
+    
+    // For details on this snippet, see: https://developer.apple.com/documentation/avfoundation/avaudiosession#
+    NSError *error;
+    [[AVAudioSession sharedInstance] setCategory:AVAudioSessionCategoryPlayback
```
Remove the audio session logic

AmitaiB replied:
✅

karimMourra commented on JWPlayer-SDK-iOS-Demo/ObjCViewController.m:
```diff
@@ -0,0 +1,46 @@
+{
+    [self.playerContainerView addSubview:playerView];
+    playerView.frame = self.playerContainerView.bounds;
+    playerView.autoresizingMask = UIViewAutoresizingFlexibleWidth & UIViewAutoresizingFlexibleHeight;
```
i don't think we need these autoresizing masks with the new fullscreen changes

karimMourra replied:
Remove these

AmitaiB replied:
@karimMourra Those changes go live in 3.3.0. Should I wait until Friday?

karimMourra commented on JWPlayer-SDK-iOS-Demo/ObjCViewController.m:
```diff
@@ -0,0 +1,46 @@
+    [self layoutPlayerView:self.player.view];
+}
+
+    - (void)viewWillDisappear:(BOOL)animated
```
remove the indentation

AmitaiB replied:
✅

karimMourra commented on JWPlayer-SDK-iOS-Demo/ObjCViewController.m:
```diff
@@ -0,0 +1,46 @@
+
+    - (void)viewWillDisappear:(BOOL)animated
+    {
+        [self.player pause];
```
remove this override, this app should only setup.

AmitaiB replied:
✅

karimMourra reviewed (changes requested):
This demo app still has unnecessary logic. Please remove. The app should setup a player, add it to the view, and do nothing else.

AmitaiB committed:
Incorporates comments

1) Removes setting AVAUdioSessionCategory.
2) Removes viewWillDisappear code.
3) Adds conditional to anticipate autolayout support.

karimMourra commented on JWPlayer-SDK-iOS-Demo/ObjCViewController.m:
```diff
@@ -0,0 +1,45 @@
+{
+    [self.playerContainerView addSubview:playerView];
+    
+    if (JWPlayerController.supportsAutolayout) {
```
This should not extend JWPlayerController; i think it would cause confusion for a developer. This is not functionality that a player controller should have.

AmitaiB replied:
Removed.

karimMourra commented on JWPlayer-SDK-iOS-Demo/HelperExtensions.swift:
```diff
@@ -0,0 +1,42 @@
+
+extension JWPlayerController {
+    /// Autolayout support was introduced in SDK v3.3.0
+    @objc static var supportsAutolayout: Bool {
```
@haawa799 please provide @AmitaiB  with feedback on how to best add the views

AmitaiB replied:
I used the autolayout pattern in `JWFreewheelVideoBaseView.m`.

karimMourra commented on JWPlayer-SDK-iOS-Demo/HelperExtensions.swift:
```diff
@@ -0,0 +1,42 @@
+
+import Foundation
+
+extension JWPlayerController {
```
@AmitaiB do not extend the player controller. This logic should not be on the player controller, and it would be confusing to developers since at a first glance it appears that this property is exposed by the player controller.

AmitaiB replied:
I'll just build it as if Autolayout works, and it'll be buggy until we release 3.3.0.

karimMourra commented on JWPlayer-SDK-iOS-Demo/ObjCViewController.m:
```diff
@@ -0,0 +1,45 @@
+    
+    if (JWPlayerController.supportsAutolayout) {
+        [playerView constrainToSuperview];
+    } else {
```
@haawa799 please advise @AmitaiB  how to best handle different layouts.

AmitaiB committed:
Presumes Autolayout support, removes complex logic

Incorporates comments

AmitaiB commented on JWPlayer-SDK-iOS-Demo/SwiftViewController.swift:
```diff
@@ -0,0 +1,51 @@
+                                                                   options: [],
+                                                                   metrics: nil,
+                                                                   views: ["thisView": self])
+        
```
@karimMourra This is the pattern used in {{JWFreewheelVideoBaseView.m}}, line ~162.

AmitaiB committed:
Updates SDK to 3.3.0; simplifies Endpoints, cleans up JWPFactory
